core/services/fluxmonitor/flux_monitor_test.go

```diff
@@ -300,6 +300,82 @@ func TestPollingDeviationChecker_TriggerIdleTimeThreshold(t *testing.T) {
 	}
 }
 
+func TestPollingDeviationChecker_RoundTimeoutCausesPoll(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	nodeAddr := ensureAccount(t, store)
+
+	tests := []struct {
+		name              string
+		timesOutAt        int64
+		expectedToTrigger bool
+	}{
+		{"timesOutAt == 0", 0, false},
+		{"timesOutAt != 0", time.Now().Add(1 * time.Second).Unix(), true},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fetcher := new(mocks.Fetcher)
+			runManager := new(mocks.RunManager)
+			fluxAggregator := new(mocks.FluxAggregator)
+
+			job := cltest.NewJobWithFluxMonitorInitiator()
+			initr := job.Initiators[0]
+			initr.ID = 1
+			initr.PollingInterval = models.Duration(math.MaxInt64)
+			initr.IdleThreshold = models.Duration(0)
+
+			const fetchedAnswer = 100
+			answerBigInt := big.NewInt(fetchedAnswer * int64(math.Pow10(int(initr.InitiatorParams.Precision))))
+
+			fluxAggregator.On("SubscribeToLogs", mock.Anything).Return(true, eth.UnsubscribeFunc(func() {}), nil)
+
+			if test.expectedToTrigger {
+				fluxAggregator.On("RoundState", nodeAddr).Return(contracts.FluxAggregatorRoundState{
+					ReportableRoundID: 1,
+					EligibleToSubmit:  false,
+					LatestAnswer:      answerBigInt,
+					TimesOutAt:        uint64(test.timesOutAt),
+				}, nil).Once()
+				fluxAggregator.On("RoundState", nodeAddr).Return(contracts.FluxAggregatorRoundState{
+					ReportableRoundID: 1,
+					EligibleToSubmit:  false,
+					LatestAnswer:      answerBigInt,
+					TimesOutAt:        0,
+				}, nil).Once()
+			} else {
+				fluxAggregator.On("RoundState", nodeAddr).Return(contracts.FluxAggregatorRoundState{
+					ReportableRoundID: 1,
+					EligibleToSubmit:  false,
+					LatestAnswer:      answerBigInt,
+					TimesOutAt:        uint64(test.timesOutAt),
+				}, nil).Once()
+			}
```

Comment:
This functions getting complicated, I personaly would prefer two tests instead of one with this if statement

Reply:
Fair point, but I'd argue that this makes it easier to compare expected behavior under different conditions.

Reply:
I dunno, I find this really hard to grok.  I'd speculate over the future the team will spend quite a lot of time collectively trying to understand and possibly update this portion of code.

I may try another sweep and improving test clarity, could include this in it.

Reply:
Sure, let's make a new story/PR for that -- we should take some time gathering feedback from others about things like this.